### PR TITLE
Set appOptions.dialog in helper instead of view

### PR DESCRIPTION
**Pull request by balderdash**
appOptions.dialog.autoResizeScrollableElement is never set to anything else, so I just hardcoded '.scrollable-element' in dialogHelper.js and moved the rest to levels_helper.rb.

Moving everything into one place makes it easier to generate multiple versions of appOptions to support CYOA.

### CODE REVIEW

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -182,6 +182,17 @@ def app_options
       # currently, all levels are Blockly or DSLDefined except for Unplugged
       @app_options = view_options.camelize_keys
     end
+
+    @app_options[:dialog] = {
+      skipSound: !!(@level.properties['options'].try(:[], 'skip_sound')),
+      preTitle: @level.properties['pre_title'],
+      fallbackResponse: @fallback_response.to_json,
+      callback: @callback,
+      app: @level.type.underscore,
```
Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb:191]: is this how app was being defined previously?

Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb:191]: Also, where are `@fallback_response` and `@callback` defined?

**Comment by Hamms** [comment]
LGTM other than comments

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (balderdash) [inline dashboard/app/helpers/levels_helper.rb:191]: The values were just hardcoded in a bunch of views, but they all follow this format. According to code-studio/src/js/reporting.js, `app` is supposed to be "The app name, as defined by its model." 

`@fallback_response` and `@callback` are defined in `ScriptLevelsController#present_level`.

Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb:191]: I'm a bit nervous about that usage for `app`; there are some strange gotchas in our set of app names.

Comment (balderdash) [inline dashboard/app/helpers/levels_helper.rb:191]: Hmm, I'll look a bit closer then. Maybe I'll just keep using the hardcoded values but take it out of appOptions.dialog.

Comment (balderdash) [inline dashboard/app/helpers/levels_helper.rb:191]: Alright, I'm still pretty confident that the app names are the same, and it also looks like this property is only used for milestone posts anyway, where it just gets logged.

Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb:191]: I see a lot of "turtle" in the milestone log, as well as "Jigsaw", neither of which types appear in my local database

Comment (balderdash) [inline dashboard/app/helpers/levels_helper.rb:191]: Ok. so there's another set of app names defined in the various <app>.js files under apps/src/... when they fire their own milestone posts, that's where turtle, Jigsaw, and most of the rest of the names actually come from.

The app name I'm setting above is only actually used for milestone posts coming from external, multi, match, free response, and levelgroup levels.

Comment (Hamms) [inline dashboard/app/helpers/levels_helper.rb:191]: Ridiculous. This LGTM, then, although if you want to document your findings in a brief comment here, you might be able to save someone a similar headache in the future.

**Pull request merged**